### PR TITLE
fix(Android): fix multiple padding related issues on Android (#5150, #5152, #5153, #5154)

### PR DESCRIPTION
**Pull request by shoamano83**
### Does any other open PR do the same thing?

No

### What issue is this PR fixing?

- https://github.com/react-native-maps/react-native-maps/issues/5150
- https://github.com/react-native-maps/react-native-maps/issues/5152
- https://github.com/react-native-maps/react-native-maps/issues/5153
- https://github.com/react-native-maps/react-native-maps/issues/5154

### How did you test this PR?

I applied this change to my sample apps

- https://github.com/shoamano83/MapPaddingSample/tree/android_map_padding_on_map_ready
- https://github.com/shoamano83/MapPaddingSample/tree/android_initial_camera_and_map_padding
- https://github.com/shoamano83/MapPaddingSample/tree/android_map_padding_updated
- https://github.com/shoamano83/MapPaddingSample/tree/android_extra_padding

to confirm that padding now works on Android just same as iOS.

### CODE REVIEW

File: android/src/main/java/com/rnmaps/maps/MapManager.java [hunk 1]
```diff
@@ -193,7 +193,6 @@ public void setMapPadding(MapView view, @Nullable ReadableMap padding) {
         }
 
         view.applyBaseMapPadding(left, top, right, bottom);
-        view.map.setPadding(left, top, right, bottom);
```
Comment (shoamano83) [inline android/src/main/java/com/rnmaps/maps/MapManager.java:196]: `map.setPadding()` is already called during `applyBaseMapPadding()` so this line seems unnecessary.

**Comment by shoamano83** [comment]
I tried to split the change into multiple PRs but there were quite a few conflicting changes and it was difficult. Please allow me to consolidate the changes into a single PR.

It seems that `com.google.android.gms.maps.GoogleMap` doesn't reapply camera after `setPadding()` is invoked. (On iOS, it does.) This is the root cause of issues https://github.com/react-native-maps/react-native-maps/issues/5150, https://github.com/react-native-maps/react-native-maps/issues/5152 and https://github.com/react-native-maps/react-native-maps/issues/5153.

So,
- when we set up camera and padding, we should make sure to call `setPadding()` first, followed by `moveCamera()` / `animateCamera()`.
- when we update `mapPadding` property, we should reapply camera for the padding to work. (Otherwise, only Google logo moves.)

I also created a patch which can be applied by `patch-package`. Let me know if you have any feedback. Thanks!
[react-native-maps+1.18.0.patch](https://github.com/user-attachments/files/16791841/react-native-maps%2B1.18.0.patch)

**Comment by shoamano83** [comment]
It seems the fix for issue https://github.com/react-native-maps/react-native-maps/issues/5152 isn't perfect. Let me work on this PR a little more.

**Comment by shoamano83** [comment]
OK I fixed the issue and recreated this PR. It is review ready now.

Here's the patch file which can be applied by `patch-package`. It should also apply cleanly against v1.15.4 (for those who need to stick with older React Native release.)
[react-native-maps+1.18.0.patch](https://github.com/user-attachments/files/16817838/react-native-maps%2B1.18.0.patch)

**Comment by shoamano83** [comment]
Hi @salah-ghanim, this PR is review ready. May I ask for a review? Thanks!

**Comment by salah-ghanim** [comment]
@shoamano83 thanks for the submission, I will soon start reviewing the bugs reported and the pull request

**Comment by shoamano83** [comment]
Hi @salah-ghanim, thank you for reviewing this PR!

My team uses this patch for a while and found an issue. Let me share it:
```
When `mapPadding` prop of a MapView is updated after a user drag the map, its camera is reset to `initialCamera`.

Steps to repro:
1. Set up a MapView with `initialCamera` prop
2. Drag the map to a different position
3. Update `mapPadding` prop of the MapView
```
The issue is caused by `lastCamera` introduced by this PR. I'm working on a fix.

**Comment by shoamano83** [comment]
> ```
> When `mapPadding` prop of a MapView is updated after a user drag the map, its camera is reset to `initialCamera`.
> 
> Steps to repro:
> 1. Set up a MapView with `initialCamera` prop
> 2. Drag the map to a different position
> 3. Update `mapPadding` prop of the MapView
> ```
> 
> The issue is caused by `lastCamera` introduced by this PR. I'm working on a fix.

I updated this PR to address this issue.

Highlights of commit f79f218:
- I reverted codes related to `lastCamera`.
- `applyBaseMapPadding()` includes extra logic to retrieve current camera (with base and edge paddings applied) and restore it after padding is updated.

I verified that with this change, issues #5150, #5152, #5153 and #5154 are still resolved.

Please kindly review again. Thank you!

**Review by salah-ghanim — APPROVED** [review]

**Pull request merged**